src/platform/Linux/ThreadStackManagerImpl.cpp

```diff
@@ -222,39 +235,48 @@ CHIP_ERROR ThreadStackManagerImpl::_SetThreadEnabled(bool val)
 
     if (val)
     {
-        std::vector<uint8_t> masterkey(std::begin(mNetworkInfo.ThreadMasterKey), std::end(mNetworkInfo.ThreadMasterKey));
-        std::vector<uint8_t> pskc;
-        uint64_t extPanId    = UINT64_MAX;
-        uint32_t channelMask = UINT32_MAX;
-
-        if (mNetworkInfo.FieldPresent.ThreadExtendedPANId)
-        {
-            extPanId = 0;
-            for (size_t i = 0; i < extPanId; i++)
-            {
-                extPanId <<= CHAR_BIT;
-                extPanId |= mNetworkInfo.ThreadExtendedPANId[i];
-            }
-        }
-        if (mNetworkInfo.FieldPresent.ThreadPSKc)
+        if (mOperationalDatasetTlv.size() > 0)
         {
-            pskc = std::vector<uint8_t>(std::begin(mNetworkInfo.ThreadPSKc), std::end(mNetworkInfo.ThreadPSKc));
+            SuccessOrExit(error = mThreadApi->SetActiveDatasetTlvs(mOperationalDatasetTlv));
+            SuccessOrExit(error = mThreadApi->Attach(
+                              [](ClientError result) { ChipLogProgress(DeviceLayer, "Thread attach result %d", result); }));
```

Comment:
Shall we post the event after the attach really succeeds?

Reply:
I guess this is achieved by "_ThreadDevcieRoleChangedHandler"